src/main/java/de/tum/in/www1/artemis/web/rest/hestia/ExerciseHintResource.java

```diff
@@ -168,20 +174,25 @@ public ResponseEntity<String> getHintTitle(@PathVariable Long exerciseId, @PathV
     }
 
     /**
-     * {@code GET  exercises/:exerciseId/exercise-hints/:exerciseHintId} : get the exerciseHint with the given id.
+     * {@code GET programming-exercises/:exerciseId/exercise-hints/:exerciseHintId} : get the exerciseHint with the given id.
      *
      * @param exerciseHintId the id of the exerciseHint to retrieve.
-     * @param exerciseId the exerciseId of the exercise of which to retrieve the exerciseHint
+     * @param exerciseId     the exerciseId of the exercise of which to retrieve the exerciseHint
      * @return the {@link ResponseEntity} with status {@code 200 (OK)} and with body the exerciseHint,
      * or with status {@code 404 (Not Found)},
      * or with status {@code 409 (Conflict)} if the exerciseId is not valid.
      */
-    @GetMapping("exercises/{exerciseId}/exercise-hints/{exerciseHintId}")
-    @PreAuthorize("hasRole('EDITOR')")
+    @GetMapping("programming-exercises/{exerciseId}/exercise-hints/{exerciseHintId}")
+    @PreAuthorize("hasRole('TA')")
     public ResponseEntity<ExerciseHint> getExerciseHint(@PathVariable Long exerciseId, @PathVariable Long exerciseHintId) {
         log.debug("REST request to get ExerciseHint : {}", exerciseHintId);
         ProgrammingExercise exercise = programmingExerciseRepository.findByIdElseThrow(exerciseId);
-        authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.EDITOR, exercise, null);
+        if (exercise.isExamExercise()) {
+            // not allowed for exam exercises
+            throw new AccessForbiddenException("");
```

Comment:
Is that the correct exception? The access is not forbidden, but rather the exercise is not suitable for hints

Reply:
This was how it was done before already. Do you have a specific Exception in mind to use here?

Reply:
Maybe the `MethodNotAllowedException` (405)?

Reply:
Typically this would be a 400 Bad Request. The client requests the hints for an exercise that doesn't support it, thus it's the client's fault.